tests/test_spatial_avg.py

```diff
@@ -453,6 +453,79 @@ def test_returns_area_weights_for_region_within_lon(self):
 
         xr.testing.assert_allclose(result, expected)
 
+
+class TestGetLongitudeWeights:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.ds = generate_dataset(cf_compliant=True, has_bounds=True)
+
+    def test_returns_weights_for_region_in_lon(self):
+        # Longitude axes orientation swaps from (-180, 180) to (0, 360).
+        result = self.ds.spatial._get_longitude_weights(
+            domain_bounds=self.ds.lon_bnds.copy(),
+            region_bounds=np.array([-170.0, -120.0]),
+        )
+
+        expected = xr.DataArray(
+            data=np.array([0.0, 0.0, 50.0, 0.0]),
+            coords={"lon": self.ds.lon},
+            dims=["lon"],
+        )
+
+        xr.testing.assert_allclose(result, expected)
+
+    def test_weights_for_region_in_lon_domain_with_both_spanning_p_meridian(self):
+        ds = self.ds.copy()
+        # Domain spans prime meridian.
+        ds.lon_bnds.data[:] = np.array([[359, 1], [1, 90], [90, 180], [180, 359]])
+
+        result = ds.spatial._get_longitude_weights(
+            domain_bounds=ds.lon_bnds,
+            # Region spans prime meridian.
+            region_bounds=np.array([359, 1]),
+        )
+        expected = xr.DataArray(
+            data=np.array([2, 0.0, 0.0, 0.0]),
+            coords={"lon": ds.lon},
+            dims=["lon"],
+        )
+
+        xr.testing.assert_allclose(result, expected)
+
+    def test_weights_for_region_in_lon_domain_with_domain_spanning_p_meridian(self):
+        ds = self.ds.copy()
+        # Domain spans prime meridian.
+        ds.lon_bnds.data[:] = np.array([[359, 1], [1, 90], [90, 180], [180, 359]])
+
+        # Longitude axes orientation swaps from (-180, 180) to (0, 360).
+        result = ds.spatial._get_longitude_weights(
+            domain_bounds=ds.lon_bnds,
+            region_bounds=np.array([-170.0, -120.0]),
+        )
+        expected = xr.DataArray(
+            data=np.array([0.0, 0.0, 0.0, 50.0]),
+            coords={"lon": ds.lon},
+            dims=["lon"],
+        )
+
+        xr.testing.assert_allclose(result, expected)
+
+    def test_weights_for_region_in_lon_domain_with_region_spanning_p_meridian(self):
+        ds = self.ds.copy()
+
+        result = ds.spatial._get_longitude_weights(
+            domain_bounds=ds.lon_bnds,
+            # Region spans prime meridian.
+            region_bounds=np.array([359, 1]),
+        )
+        expected = xr.DataArray(
+            data=np.array([1.875, 0.0625, 0.0, 0.0625]),
+            coords={"lon": ds.lon},
+            dims=["lon"],
+        )
+
+        xr.testing.assert_allclose(result, expected)
```

Comment:
@pochedls Can you review these tests to confirm that the expected is actually the expected?

Reply:
This looks good to me.